utils/fetch_currencies.py

```diff
@@ -1,163 +1,139 @@
+        iso4217 = r['iso4217']['value']
+        article_name = r['article_name']['value']
+        article_lang = r['article_name']['xml:lang']
+        add_currency_name(db, article_name, iso4217)
+        add_currency_label(db, article_name, iso4217, article_lang)
 
+    for r in wikidata_request_result_iterator(SARQL_REQUEST):
+        iso4217 = r['iso4217']['value']
+        if 'label' in r:
+            label = r['label']['value']
+            label_lang = r['label']['xml:lang']
+            add_currency_name(db, label, iso4217)
+            add_currency_label(db, label, iso4217, label_lang)
 
-def fetch_data_batch(wikidata_ids):
-    while len(wikidata_ids) > 0:
-        if len(wikidata_ids) > 50:
-            fetch_data(wikidata_ids[0:49])
-            wikidata_ids = wikidata_ids[50:]
-        else:
-            fetch_data(wikidata_ids)
-            wikidata_ids = []
+        if 'alias' in r:
+            add_currency_name(db, r['alias']['value'], iso4217)
 
+        if 'unicode' in r:
+            add_currency_name(db, r['unicode']['value'], iso4217, normalize_name=False)
 
-def wdq_query(query):
-    url = url_wmflabs_template + query
-    htmlresponse = get(url)
-    jsonresponse = json.loads(htmlresponse.content)
-    qlist = list(map(add_q, jsonresponse.get('items', {})))
-    error = jsonresponse.get('status', {}).get('error', None)
-    if error is not None and error != 'OK':
-        print("error for query '" + query + "' :" + error)
+        if 'unit' in r:
+            add_currency_name(db, r['unit']['value'], iso4217, normalize_name=False)
 
-    fetch_data_batch(qlist)
+    for name in db['names']:
+        if len(db['names'][name]) == 1:
+            db['names'][name] = db['names'][name][0]
 
+    return db
 
-def wd_query(query, offset=0):
-    qlist = []
 
-    url = url_wikidata_search_template.format(query=urlencode({'srsearch': query, 'srlimit': 50, 'sroffset': offset}))
-    htmlresponse = get(url)
-    jsonresponse = json.loads(htmlresponse.content)
-    for r in jsonresponse.get('query', {}).get('search', {}):
-        qlist.append(r.get('title', ''))
-    fetch_data_batch(qlist)
+def get_filename():
+    return join(join(searx_dir, "data"), "currencies.json")
 
 
-# fetch #
-for q in wmflabs_queries:
-    wdq_query(q)
+def main():
+    #
+    db = fetch_db()
+    # static
+    add_currency_name(db, "euro", 'EUR')
+    add_currency_name(db, "euros", 'EUR')
+    add_currency_name(db, "dollar", 'USD')
+    add_currency_name(db, "dollars", 'USD')
+    add_currency_name(db, "peso", 'MXN')
+    add_currency_name(db, "pesos", 'MXN')
 
-# static
-add_currency_name("euro", 'EUR')
-add_currency_name("euros", 'EUR')
-add_currency_name("dollar", 'USD')
-add_currency_name("dollars", 'USD')
-add_currency_name("peso", 'MXN')
-add_currency_name("pesos", 'MXN')
+    with open(get_filename(), 'w', encoding='utf8') as f:
+        json.dump(db, f, ensure_ascii=False, separators=(',', ':'))
```

Comment:
As long as we commit data files in the SCM we should make them [diffable](https://en.wiktionary.org/wiki/diffable), I would suggest. Here is my patch:
```diff
@@ -133,7 +133,7 @@ def main():
     add_currency_name(db, "pesos", 'MXN')
 
     with open(get_filename(), 'w', encoding='utf8') as f:
-        json.dump(db, f, ensure_ascii=False, separators=(',', ':'))
+        json.dump(db, f, ensure_ascii=False, separators=(',', ':'), sort_keys=True, indent=2)
 ```

I choose a indent of 2 to save some space, but 4 is more common / delta is about ~70kB.  But I don't know if we should care about the size (compare `searx/data/bangs.json`)

Reply:
Updated.